src/slapp.js

```diff
@@ -120,7 +133,25 @@ class Slapp extends EventEmitter {
   }
 
   /**
-   * Middleware that ignores messages from any bot user when we can tell
+   * Middleware that ignores messages from this bot user (self) when we can tell. Requires the
+   * meta context to be populated with `app_bot_id`.
+   *
+   * @api private
+   */
+
+  ignoreSelfMiddleware () {
+    return (msg, next) => {
+      if (msg.isBot() && msg.isMessage() && msg.body.event.subtype === 'bot_message') {
+        if (msg.meta.app_bot_id === msg.meta.bot_id) {
```

Comment:
Should we test that one of these isn't falsey too?  We do that in some other places to try and avoid a false positive where both are `undefined` or `null`

```
if (msg.meta.app_bot_id && msg.meta.app_bot_id === msg.meta.bot_id) {
```